### PR TITLE
closes #308: Extract config for google maps key

### LINKED ISSUE #308: Extract code that hardcodes the google maps API key into a config/env-var
DefaultGeocodeClient currently has a hardcoded client ID.

Push this into config.

### PR DESCRIPTION
**Pull request by RyanBard**
* Adds Thymeleaf dependency for html templates.
* Adds config to application.properties for google maps api key.
* Adds config to PublicConfig for google maps api key.
* Changes static/index.html to be a template and replaces hardcoded maps key
  with configured maps key.
* Replaces hardcoded maps key in DefaultGoogleGeocodeClient with configured
  maps key.
* Adds unit tests for ViewsController.

### CODE REVIEW

File: config/application.properties [hunk 1]
```diff
@@ -1,6 +1,18 @@
 logging.path = logs/
 
 #####################################################################
+# Begin: Google maps configuration
+#####################################################################
+
+google.geocode.client.id = ${GOOGLE_GEOCODE_CLIENT_ID}
```
Comment (RyanBard) [inline config/application.properties]: This is now required in your .env to run locally (and you'll have to add it to your config in heroku as well -- I did this for hmhb, hmhb-ryan, and hmhb-demo already, but I don't have access to hmhb-mckoon).

File: src/main/resources/templates/index.html [hunk 2]
```diff
@@ -1,39 +1,41 @@
 <!DOCTYPE html>
 
 <html lang="en"
+      xmlns:th="http://www.thymeleaf.org"
       ng-app="healthBam">
 
     <head>
 
-        <meta charset="UTF-8">
+        <meta charset="UTF-8" />
```
Comment (RyanBard) [inline src/main/resources/templates/index.html]: Thymeleaf is very complainful :)

It wants xml (or maybe xhtml), so I had to make sure all tags were closed and all attributes had a =.

File: src/test/java/org/hmhb/ViewsControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import org.springframework.core.env.Environment;
+import org.springframework.ui.Model;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+/**
+ * Unit tests for {@link ViewsController}.
+ */
+public class ViewsControllerTest {
```
Comment (RyanBard) [inline src/test/java/org/hmhb/ViewsControllerTest.java]: I noticed this test was missing, so I added it.

File: config/application.properties (hunk 1 above)
Comment (mckoon) [inline config/application.properties]: Added to hmhb-mckoon and my .env.

**Pull request merged**